FIX: Song's Overlay with Fast Ocarina Playback

PurpleHato opened the pull request:
Fixes https://github.com/HarbourMasters/Shipwright/issues/1773

Before:
![image](https://user-images.githubusercontent.com/47987542/217005411-00f1b566-7f92-45a6-9de5-8f17441f1ae7.png)

After:
![image](https://user-images.githubusercontent.com/47987542/217005452-e81414b6-1c4a-4821-ad99-2a881ca88bc7.png)


<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux-compatibility.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630798.zip)
  - [soh-linux-performance.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630799.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630800.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630801.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630802.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/543630803.zip)
<!--- section:artifacts:end -->

PurpleHato committed:
FIX: Affected songs

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe/z_oceff_wipe.c:
```diff
@@ -80,10 +80,12 @@ void OceffWipe_Draw(Actor* thisx, PlayState* play) {
 
     OPEN_DISPS(play->state.gfxCtx);
 
+    int fastOcarinaPlayback = CVarGetInteger("gFastOcarinaPlayback", 0);
```
```suggestion
    int fastOcarinaPlayback = (CVarGetInteger("gFastOcarinaPlayback", 0) != 0);
```
doing this will make it always 1 or zero, so we can treat it as a bool later (realistically we don't need to do this, as we should only ever get a 1 or zero from the cvar)

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe/z_oceff_wipe.c:
```diff
@@ -80,10 +80,12 @@ void OceffWipe_Draw(Actor* thisx, PlayState* play) {
+    int fastOcarinaPlayback = CVarGetInteger("gFastOcarinaPlayback", 0);
     if (this->timer < 32) {
-        z = Math_SinS(this->timer << 9) * 1400;
+        float sinVal = Math_SinS(this->timer << 9);
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1400.0f;
```
```suggestion
        z = Math_SinS(this->timer << 9) * (fastOcarinaPlayback ? 1200.0f : 1400.0f);
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe/z_oceff_wipe.c:
```diff
@@ -80,10 +80,12 @@ void OceffWipe_Draw(Actor* thisx, PlayState* play) {
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1400.0f;
     } else {
-        z = 1400;
+        z = (fastOcarinaPlayback != 0) ? 1200.0f : 1400.0f;
```
```suggestion
        z = fastOcarinaPlayback ? 1200.0f : 1400.0f;
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe2/z_oceff_wipe2.c:
```diff
@@ -71,10 +71,13 @@ void OceffWipe2_Draw(Actor* thisx, PlayState* play) {
+    int fastOcarinaPlayback = CVarGetInteger("gFastOcarinaPlayback", 0);
     if (this->timer < 32) {
-        z = Math_SinS(this->timer << 9) * 1330;
+        float sinVal = Math_SinS(this->timer << 9);
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
```
```suggestion
        z = Math_SinS(this->timer << 9) * (fastOcarinaPlayback ? 1200.0f : 1330.0f);
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe2/z_oceff_wipe2.c:
```diff
@@ -71,10 +71,13 @@ void OceffWipe2_Draw(Actor* thisx, PlayState* play) {
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
     } else {
-        z = 1330;
+        z = (fastOcarinaPlayback != 0) ? 1200.0f : 1330.0f;
```
```suggestion
        z = fastOcarinaPlayback ? 1200.0f : 1330.0f;
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe3/z_oceff_wipe3.c:
```diff
@@ -72,10 +72,13 @@ void OceffWipe3_Draw(Actor* thisx, PlayState* play) {
+    int fastOcarinaPlayback = CVarGetInteger("gFastOcarinaPlayback", 0);
     if (this->counter < 32) {
-        z = Math_SinS(this->counter << 9) * 1330;
+        float sinVal = Math_SinS(this->counter << 9);
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
```
```suggestion
        z = Math_SinS(this->counter << 9) * (fastOcarinaPlayback ? 1200.0f : 1330.0f);
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe3/z_oceff_wipe3.c:
```diff
@@ -72,10 +72,13 @@ void OceffWipe3_Draw(Actor* thisx, PlayState* play) {
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
     } else {
-        z = 1330;
+        z = (fastOcarinaPlayback != 0) ? 1200.0f : 1330.0f;
```
```suggestion
        z = fastOcarinaPlayback ? 1200.0f : 1330.0f;
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4.c:
```diff
@@ -67,10 +67,13 @@ void OceffWipe4_Draw(Actor* thisx, PlayState* play) {
+    int fastOcarinaPlayback = CVarGetInteger("gFastOcarinaPlayback", 0);
     if (this->timer < 16) {
-        z = Math_SinS(this->timer * 1024) * 1330.0f;
+        float sinVal = Math_SinS(this->timer * 1024);
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
```
```suggestion
        z = Math_SinS(this->timer * 1024) * (fastOcarinaPlayback ? 1200.0f : 1330.0f);
```

briaguya0 commented on soh/src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4.c:
```diff
@@ -67,10 +67,13 @@ void OceffWipe4_Draw(Actor* thisx, PlayState* play) {
+        z = (fastOcarinaPlayback != 0) ? sinVal * 1200.0f : sinVal * 1330.0f;
     } else {
-        z = 1330.0f;
+        z = (fastOcarinaPlayback != 0) ? 1200.0f : 1330.0f;
```
```suggestion
        z = fastOcarinaPlayback ? 1200.0f : 1330.0f;
```

briaguya0 reviewed (commented):
overall super excited to have this change in here

couple suggestions to clean up the code a little, only other thing i can think of is maybe replacing the magic numbers with `#define`d values or something but that probably isn't needed for now

PurpleHato committed:
TWEAK: Addressed comments

PurpleHato commented:
> overall super excited to have this change in here
> 
> couple suggestions to clean up the code a little, only other thing i can think of is maybe replacing the magic numbers with `#define`d values or something but that probably isn't needed for now

Addressed the comments :)
For the define and magic numbers, I think it will have it's place for when we'll fix the bigger than 16/9 screen ratio issue since we will ultimately will need a rework for these number at that time
